test/python/primitives/test_utils.py

```diff
@@ -83,3 +91,31 @@ def test_mapping_w_delays(self):
 
         maps = final_measurement_mapping(qc)
         self.assertDictEqual(maps, {1: 0, 0: 1})
+
+
+BACKENDS = [FakeNairobi(), FakeNairobiV2()]
+
+
+@ddt
+class TestTranspileOperator(QiskitTestCase):
+    """Test transpile_operator utility function."""
+
+    @combine(backend=BACKENDS)
+    def test_tranpile_operator(self, backend):
+        """test for transpile_operator"""
+        backend.set_options(seed_simulator=15)
+        n = 6
+        qc = QuantumCircuit(n)
+        qc.x(n - 1)
+        qc.h(range(n))
+        qc.cx(range(n - 1), n - 1)
+        qc.h(range(n - 1))
+        trans_qc = transpile(qc, backend, seed_transpiler=15)
+
+        op = SparsePauliOp("Z" * n)
+        trans_op = transpile_operator(op, trans_qc.layout, qc.qubits)
+        result = BackendEstimator(backend=backend).run(trans_qc, trans_op).result()
+        if optionals.HAS_AER:
+            self.assertAlmostEqual(result.values[0], -0.045, places=2)
+        else:
+            self.assertAlmostEqual(result.values[0], 0, places=2)
```

Comment:
Could you add an additional test that checks that `trans_op` equals the expected operator? Testing that the values are correct is also good, but there are other steps included which could affect the test, so I think it would be good to have a very specific one too 🙂